drivers/serial/uart_nrfx_uarte.c

```diff
@@ -1471,49 +1477,33 @@ static int uarte_nrfx_pm_control(struct device *dev, u32_t ctrl_command,
 			};))
 
 #ifdef CONFIG_UART_0_NRF_UARTE
```

Comment:
You don't use _CONFIG_RTS_CTS anymore from what I can tell, can be removed.

Reply:
It's used to set .pselcts and .pselrts.

Reply:
ah, missed that.  Can't you just use `UARTE_RTS_CTS_PINS_SET` there?

Reply:
That doesn't work with COND_CODE_1

Reply:
But instead of `UARTE_NRF_RTS_CTS_PINS(idx)` you could use:
```
.pselrts = UARTE_RTS_PIN(idx),
.pselcts = UARTE_CTS_PIN(idx),
```
with `UARTE_RTS_PIN` and `UARTE_CTS_PIN` implemented just like `UARTE_RX_PIN`. These fields are accessed only in one [block](https://github.com/zephyrproject-rtos/zephyr/blob/8bf348d2a2ad184bacd93d54b024f7fa85be2998/drivers/serial/uart_nrfx_uarte.c#L1205) that is guarded by an `if` checking if both pins are configured, so there is no need to check both pins also in `COND_CODE_1` conditions.

Then you could get rid of `UARTE_x_CONFIG_RTS_CTS`.

Reply:
> Then you could get rid of UARTE_x_CONFIG_RTS_CTS.

Good idea, now done.